tests/playwright/src/model/pages/create-machine-page.ts

```diff
@@ -0,0 +1,134 @@
+      name: 'User mode networking',
+    });
+    this.startNowCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', { name: 'Start the machine now' });
+    this.closeButton = this.page.getByRole('button', { name: 'Close' });
+    this.createMachineButton = this.page.getByRole('button', { name: 'Create' });
+  }
+
+  async createMachine(
+    machineName: string,
+    isRootful: boolean = true,
+    enableUserNet: boolean = false,
+    startNow: boolean = true,
+    setAsDefault: boolean = true,
+  ): Promise<ResourcesPage> {
+    await playExpect(this.podmanMachineConfiguration).toBeVisible();
+    await this.podmanMachineName.fill(machineName);
+
+    if (isRootful !== (await this.isEnabled(this.rootPriviledgesCheckbox))) {
+      await this.switchCheckbox(this.rootPriviledgesCheckbox);
+    }
+
+    if (enableUserNet !== (await this.isEnabled(this.userModeNetworkingCheckbox))) {
+      await this.switchCheckbox(this.userModeNetworkingCheckbox);
+    }
+
+    if (startNow !== (await this.isEnabled(this.startNowCheckbox))) {
+      await this.switchCheckbox(this.startNowCheckbox);
+    }
+
+    await this.createMachineButton.click();
+
+    const successfulCreationMessage = this.page.getByText('Successful operation');
+    const goBackToResourcesButton = this.page.getByRole('button', { name: 'Go back to resources' });
+
+    await playExpect(successfulCreationMessage).toBeVisible({ timeout: 200_000 });
+    await playExpect(goBackToResourcesButton).toBeVisible();
+    await this.page.waitForTimeout(2_000); //wait for dialog to appear, in the needed cases
+    await this.handleConnectionDialog(setAsDefault);
+
+    await goBackToResourcesButton.click();
+
+    return new ResourcesPage(this.page);
+  }
+
+  async isEnabled(checkbox: Locator): Promise<boolean> {
+    await playExpect(checkbox).toBeVisible();
+    const upperElement = checkbox.locator('..').locator('..');
+    const clickableCheckbox = upperElement.getByText('Enabled');
+    return await clickableCheckbox.isVisible();
+  }
+
+  async switchCheckbox(checkbox: Locator): Promise<void> {
+    await playExpect(checkbox).toBeVisible();
+    const upperElement = checkbox.locator('..').locator('..');
+
+    const wasEnabled = await this.isEnabled(checkbox);
+    let checkText;
+    if (wasEnabled) {
+      checkText = 'Enabled';
+    } else {
+      checkText = 'Disabled';
+    }
+
+    const clickableCheckbox = upperElement.getByText(checkText);
+    await clickableCheckbox.click();
+  }
+
+  async handleConnectionDialog(setAsDefault: boolean): Promise<void> {
+    const connectionDialog = this.page.getByRole('dialog', { name: 'Podman' });
+    if (await connectionDialog.isVisible()) {
+      let handleButtonName = 'Yes';
+      if (!setAsDefault) {
+        handleButtonName = 'Ignore';
+      }
+      const handleButton = connectionDialog.getByRole('button', { name: handleButtonName });
+      await handleButton.click();
+    }
```

Comment:
This is not the right way to check this, the method isVisible() always returns immediately, it has no innate waiter, as such it can return false just as easily as true, simply because of system lag, this is very flaky behavior.